compiler/runtime/OMRRelocationRecord.cpp

```diff
@@ -0,0 +1,393 @@
+            {
+            int32_t offsetHigh = *offsetPtr;
+            int32_t offsetLow = *(offsetPtr+1);
+            uint8_t *reloLocationHigh = reloOrigin + offsetHigh + 2; // Add 2 to skip the first 16 bits of instruction
+            uint8_t *reloLocationLow = reloOrigin + offsetLow + 2; // Add 2 to skip the first 16 bits of instruction
+            int32_t rc = self()->applyRelocation(reloRuntime, reloTarget, reloLocationHigh, reloLocationLow);
+            if (rc != 0)
+               {
+               return rc;
+               }
+            }
+         }
+      else
+         {
+         int16_t *offsetsBase = (int16_t *) (((uint8_t*)_record) + self()->bytesInHeader());
+         int16_t *endOfOffsets = (int16_t *) self()->nextBinaryRecord(reloTarget);
+         for (int16_t *offsetPtr = offsetsBase;offsetPtr < endOfOffsets; offsetPtr+=2)
+            {
+            int16_t offsetHigh = *offsetPtr;
+            int16_t offsetLow = *(offsetPtr+1);
+            uint8_t *reloLocationHigh = reloOrigin + offsetHigh + 2; // Add 2 to skip the first 16 bits of instruction
+            uint8_t *reloLocationLow = reloOrigin + offsetLow + 2; // Add 2 to skip the first 16 bits of instruction
+            int32_t rc = self()->applyRelocation(reloRuntime, reloTarget, reloLocationHigh, reloLocationLow);
+            if (rc != 0)
+               {
+               return rc;
+               }
+            }
+         }
+      }
+   else
+      {
+      if (self()->wideOffsets(reloTarget))
+         {
+         int32_t *offsetsBase = (int32_t *) (((uint8_t*)_record) + self()->bytesInHeader());
+         int32_t *endOfOffsets = (int32_t *) self()->nextBinaryRecord(reloTarget);
+         for (int32_t *offsetPtr = offsetsBase;offsetPtr < endOfOffsets; offsetPtr++)
+            {
+            int32_t offset = *offsetPtr;
+            uint8_t *reloLocation = reloOrigin + offset;
+            int32_t rc = self()->applyRelocation(reloRuntime, reloTarget, reloLocation);
+            if (rc != 0)
+               {
+               return rc;
+               }
+            }
+         }
+      else
+         {
+         int16_t *offsetsBase = (int16_t *) (((uint8_t*)_record) + self()->bytesInHeader());
+         int16_t *endOfOffsets = (int16_t *) self()->nextBinaryRecord(reloTarget);
+         for (int16_t *offsetPtr = offsetsBase;offsetPtr < endOfOffsets; offsetPtr++)
+            {
+            int16_t offset = *offsetPtr;
+            uint8_t *reloLocation = reloOrigin + offset;
+            int32_t rc = self()->applyRelocation(reloRuntime, reloTarget, reloLocation);
+            if (rc != 0)
+               {
+               return rc;
+               }
+            }
+         }
+      }
+      return 0;
+   }
+   
+
+uint32_t OMR::RelocationRecord::_relocationRecordHeaderSizeTable[TR_NumExternalRelocationKinds] =
+   {
+      #include "OMRExternalRelocationsKindToHeaderSizeMap.hpp"
+   };
```

Comment:
As mentioned above, put the contents of `OMRExternalRelocationsKindToHeaderSizeMap.hpp` here; you don't need the header at all. All the codegens that need the size should be getting it via `OMR::RelocationRecord::getSizeOfAOTRelocationHeader` - ie instead of `codeGen->getAheadOfTimeCompile()->getSizeOfAOTRelocationHeader(k)`.

Reply:
Hi, thank you for comment! I will update it and upload the solution with the next commit!